src/constraint/field.ts

```diff
@@ -22,7 +23,7 @@ export const FIELD_CONSTRAINTS: EncodingConstraintModel<FieldQuery>[] = [
     strict: true,
     satisfy: (fieldQ: FieldQuery, _: Schema, __: PropIndex<Wildcard<any>>, ___: QueryConfig) => {
       if (fieldQ.aggregate) {
-        return fieldQ.type !== Type.ORDINAL && fieldQ.type !== Type.NOMINAL;
+        return fieldQ.type !== Type.ORDINAL && fieldQ.type !== Type.NOMINAL && fieldQ.type !== ExpandedType.KEY;
```

Comment:
You might wanna add `isDiscrete` to `expandedtype.ts` and use `!isDiscrete(fieldQ.type)`